### PR TITLE
[nocheck] PyDoc: fix warnings

### PR DESCRIPTION
**Pull request by sebhrusen**
Fix warnings and broken links (after refactorings in `h2o.models` module) when generating PyDoc

### CODE REVIEW

File: h2o-bindings/bin/gen_python.py [hunk 1]
```diff
@@ -405,6 +405,7 @@ def create_estimator(algo, **params):
 
 def gen_models_docs(modules):
     yield ".. This file is autogenerated from gen_python.py, DO NOT MODIFY"
+    yield ""
```
Comment (sebhrusen) [inline h2o-bindings/bin/gen_python.py:408]: removes the following warning:
```
/Users/seb/repos/h2o/h2o-3/h2o-py/docs/modeling.rst:2: WARNING: Explicit markup ends without a blank line; unexpected unindent.
```

File: h2o-py/docs/frame.rst [hunk 2]
```diff
@@ -15,9 +15,3 @@ Data Manipulation
     :undoc-members:
     :show-inheritance:
 
-:mod:`TargetEncoder`
---------------------
-.. autoclass:: h2o.targetencoder.TargetEncoder
-    :members:
-    :undoc-members:
-    :show-inheritance:
```
Comment (sebhrusen) [inline h2o-py/docs/frame.rst:23]: module was deleted long ago.
Removes the following warning:
```
WARNING: autodoc: failed to import class 'targetencoder.TargetEncoder' from module 'h2o'; the following exception was raised:
No module named 'h2o.targetencoder'
```

File: h2o-py/docs/metrics.rst [hunk 3]
```diff
@@ -9,6 +9,12 @@ Metrics in H2O
   :show-inheritance:
   :noindex:
 
+.. automodule:: h2o.model.metrics
+  :members:
+  :undoc-members:
+  :show-inheritance:
+  :noindex:
+
```
Comment (sebhrusen) [inline h2o-py/docs/metrics.rst:17]: after some refactoring, specialized metrics where moved from `h2o.model.metrics_base` module to dedicated `h2o.model.metrics.xxx`

File: h2o-py/docs/model_categories.rst [hunk 4]
```diff
@@ -14,65 +14,90 @@ Model Categories
     :undoc-members:
     :show-inheritance:
 
-:mod:`MetricsBase`
-------------------
-
-.. automodule:: h2o.model.metrics_base
-    :members:
-    :undoc-members:
-    :show-inheritance:
```
Comment (sebhrusen) [inline h2o-py/docs/model_categories.rst:23]: not sure why this is needed here as we already have this in the metrics section

Comment (sebhrusen) [reply h2o-py/docs/model_categories.rst:23]: do we want to keep it to avoid broken links?

File: h2o-py/docs/model_categories.rst [hunk 5]
```diff
@@ -14,65 +14,90 @@ Model Categories
...
 
 :mod:`Binomial Classification`
 ------------------------------
 
-.. automodule:: h2o.model.binomial
+.. automodule:: h2o.model.models.binomial
```
Comment (sebhrusen) [inline h2o-py/docs/model_categories.rst:21]: specialized methods have been refactored and moved from `h2o.model.xxx` to `h2o.model.models.xxx` (also for symmmetry with h2o.model.metrics`)

File: h2o-py/h2o/frame.py [hunk 6]
```diff
@@ -4966,6 +4966,7 @@ def show(self, use_pandas=None, rows=10, cols=200, verbosity=None, fmt=None):
         :param cols:
         
         :examples:
+
```
Comment (sebhrusen) [inline h2o-py/h2o/frame.py:4969]: changes in this file to fix warnings:
```
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/frame.py:docstring of h2o.H2OFrame.get_summary:9: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/frame.py:docstring of h2o.H2OFrame.show:9: WARNING: Field list ends without a blank line; unexpected unindent.
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/frame.py:docstring of h2o.H2OFrame.show_summary:6: WARNING: Field list ends without a blank line; unexpected unindent.
```

File: h2o-py/h2o/model/models/anomaly_detection.py [hunk 7]
```diff
@@ -10,16 +10,16 @@ class H2OAnomalyDetectionModel(ModelBase):
...
-        - aggregated_split_depths - The sum of all depths of a variable used to make a split. (If a variable is used
-                                    on level N of a tree, then it contributes with N to the total aggregate.)
+        
+        - count: The number of times a variable was used to make a split.
+        - aggregated_split_ratios: The split ratio is defined as "abs(#left_observations - #right_observations) / #before_split".
+            Even splits (#left_observations approx the same as #right_observations) contribute
+            less to the total aggregated split ratio value for the given feature;
+            highly imbalanced splits (eg. #left_observations >> #right_observations) contribute more.
+        - aggregated_split_depths: The sum of all depths of a variable used to make a split. 
+            (If a variable is used on level N of a tree, then it contributes with N to the total aggregate.)
```
Comment (sebhrusen) [inline h2o-py/h2o/model/models/anomaly_detection.py:20]: reformatted this part to fix warning:
```
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/model/models/anomaly_detection.py:docstring of h2o.model.models.anomaly_detection.H2OAnomalyDetectionModel.varsplits:5: WARNING: Unexpected indentation.
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/model/models/anomaly_detection.py:docstring of h2o.model.models.anomaly_detection.H2OAnomalyDetectionModel.varsplits:8: WARNING: Block quote ends without a blank line; unexpected unindent.
```

File: h2o-py/h2o/grid/grid_search.py [hunk 8]
```diff
@@ -1573,6 +1573,7 @@ def pareto_front(self,
         :return: object that contains the resulting figure (can be accessed using ``result.figure()``)
 
         :examples:
+
```
Comment (sebhrusen) [inline h2o-py/h2o/grid/grid_search.py:1576]: fixing warning:
```
/Users/seb/repos/h2o/h2o-3/h2o-py/h2o/grid/grid_search.py:docstring of h2o.grid.H2OGridSearch.pareto_front:16: WARNING: Field list ends without a blank line; unexpected unindent.

```

File: h2o-py/h2o/model/models/anomaly_detection.py [hunk 9]
```diff
@@ -10,16 +10,19 @@ class H2OAnomalyDetectionModel(ModelBase):
...
+        - count
+            The number of times a variable was used to make a split.
+        - aggregated_split_ratios
+            The split ratio is defined as ``abs(#left_observations - #right_observations) / #before_split``.
+            Even splits (``#left_observations`` approx the same as ``#right_observations``) contribute
+            less to the total aggregated split ratio value for the given feature;
+            highly imbalanced splits (eg. ``#left_observations >> #right_observations``) contribute more.
+        - aggregated_split_depths
+            The sum of all depths of a variable used to make a split. 
+            (If a variable is used on level N of a tree, then it contributes with N to the total aggregate.)
```
Comment (sebhrusen) [inline h2o-py/h2o/model/models/anomaly_detection.py:23]: @hannah-tillman looks like the best way to have lists with long descriptions for each item

**Review by hannah-tillman — APPROVED** [review]
LGTM :)

**Pull request merged**